src/app/util/BUILD.gn

```diff
@@ -23,6 +23,7 @@ source_set("types") {
     "attribute-metadata.h",
     "attribute-storage-null-handling.h",
     "basic-types.h",
+    "ember-server-cluster.h",
```

Comment:
Types generally should not depend on generated code. This depends on `app-common/zap-generated/callback.h` so it seems it should not be here.

Reply:
Yeah these dependencies are tricky... my goal is to have the ember adapters statically allocated based on the `MATTER_DM_*_SERVER_ENDPOINT_COUNT` without having to manually write "stub" files for each cluster.

A different possible approach would be to address this is in the zap templates directly, maybe by listing the clusters that are based on `ServerCluster` in config-data.yaml and then have the code generation do the right thing automatically.